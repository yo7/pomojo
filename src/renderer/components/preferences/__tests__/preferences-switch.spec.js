import {shallow} from 'vue-test-utils'
import {createRenderer} from 'vue-server-renderer'
import PreferencesSwitch from '../preferences-switch.vue'

describe('PreferencesSwitch', () => {
  const onChanged = jest.fn()
  const wrapper = shallow(PreferencesSwitch, {propsData: {
    checked: true,
    onChanged
  }})

  it('switch on clicked', () => {
    const check = wrapper.find('.switch')
    check.trigger('click')
    expect(onChanged).toHaveBeenCalled()
  })

  it('renders correctly', () => {
    const renderer = createRenderer()
    renderer.renderToString(wrapper.vm, (err, str) => {
      if (err) {
        throw new Error()
      }
      expect(str).toMatchSnapshot()
    })
  })
})
