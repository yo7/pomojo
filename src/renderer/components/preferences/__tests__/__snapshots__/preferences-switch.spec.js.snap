// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PreferencesSwitch renders correctly 1`] = `<span><input type="checkbox" id="switch" checked="checked" class="switch"> <label for="switch" class="label"></label></span>`;
